### PR TITLE
Add setup

### PR DESCRIPTION
**Pull request by CunliangGeng**
To solve #22. 

Instead of using jupyter notebook, I recommend using jupyterlab for teaching.

### CODE REVIEW

File: learners/setup.md [hunk 1]
```diff
@@ -2,53 +2,154 @@
 title: Setup
 ---
 
-FIXME: Setup instructions live in this document. Please specify the tools and
-the data sets the Learner needs to have installed.
+The vantage6 has requirements on hardware and software to run its server and node instances. As for the detailed requirements, please refer to the [vantage6 documentation](https://docs.vantage6.ai/en/main/node/requirements.html).
```
Comment (frankcorneliusmartin) [inline learners/setup.md:5]: I would remove *the*. The vantage6 sounds a bit weird. And maybe then reformulate a little bit to: 

> vantage6 has hardware and software requirements in order to run the vantage6 node(s) and/or server.

File: learners/setup.md [hunk 2]
```diff
@@ -2,53 +2,154 @@
...
-       link to it, replacing the example.com link.
--->
-Download the [data zip file](https://example.com/FIXME) and unzip it to your Desktop
+## Software setup
 
-## Software Setup
+This lesson requires the following software to be installed on your system:
 
-::::::::::::::::::::::::::::::::::::::: discussion
+- Operating system: Ubuntu 18.04+ , MacOS Big Sur+, or Windows 10+
```
Comment (frankcorneliusmartin) [inline learners/setup.md:13]: I would opt for Ubuntu 20.04+ as 18.04 is no longer in maintenance. ( we probably should also update that in the docs)
image

Also these OSes are recommended, So I would change it to: *Recommended Operating systems*

File: learners/setup.md [hunk 3]
```diff
@@ -2,53 +2,154 @@
...
+- Docker (always latest version)
+- Conda (latest version)
+- Python
+    - Python v3.10 for vantage6 version 3.8.0 or higher
+    - Python v3.7 for other lower versions of vantage6
```
Comment (frankcorneliusmartin) [inline learners/setup.md:18]: Remove the Python 3.7 requirement, during the workshop we will work only with the newest version of vantage6.

File: learners/setup.md [hunk 4]
```diff
@@ -2,53 +2,154 @@
...
--->
-Download the [data zip file](https://example.com/FIXME) and unzip it to your Desktop
+## Software setup
 
-## Software Setup
+This lesson requires the following software to be installed on your system:
 
-::::::::::::::::::::::::::::::::::::::: discussion
+- Operating system: Ubuntu 18.04+ , MacOS Big Sur+, or Windows 10+
+- Docker (always latest version)
```
Comment (frankcorneliusmartin) [inline learners/setup.md:14]: "Docker Desktop"

File: learners/setup.md [hunk 5]
```diff
@@ -2,53 +2,154 @@
...
+### Docker
 
-Setup for different systems can be presented in dropdown menus via a `spoiler`
-tag. They will join to this discussion block, so you can give a general overview
-of the software used in this lesson here and fill out the individual operating
-systems (and potentially add more, e.g. online setup) in the solutions blocks.
+We recommend installing [Docker Desktop](https://www.docker.com/products/docker-desktop/) on your system, which is the easiest way to install Docker.
 
-:::::::::::::::::::::::::::::::::::::::::::::::::::
+You can follow the official instructions to install Docker Desktop on [MacOS](https://docs.docker.com/desktop/install/mac-install/), [Windows](https://docs.docker.com/desktop/install/windows-install/), or [Linux](https://docs.docker.com/desktop/install/linux-install/).
```
Comment (frankcorneliusmartin) [inline learners/setup.md:30]: Could you add a note here that in case the Linux user does not use *Docker Desktop* but the *Docker Engine* it should be able to run *Docker containers* without `sudo`, so checkout the post installation steps for linux: https://docs.docker.com/engine/install/linux-postinstall/

( I think linux user probably already have the Docker Engine installed, so they probably want to do it without Desktop)

File: learners/setup.md [hunk 6]
```diff
@@ -2,53 +2,154 @@
...
 
-:::::::::::::::: spoiler
+After installing the Docker Desktop, open the application, and then verify the installation by running the command in your terminal:
 
-### Windows
+```bash
+docker version
```
Comment (frankcorneliusmartin) [inline learners/setup.md:34]: Maybe let them do a `docker run hello-world` instead. The Docker CLI will work without Docker actually running, so its safer to run a `hello -world` container.

Comment (frankcorneliusmartin) [reply learners/setup.md:34]: Oh I see that you explain how to check that Docker is running in the spoiler sections, that is also OK for me. So feel free to leave it as it is.

**Review by frankcorneliusmartin — CHANGES_REQUESTED** [review]
Some minor findings. Also, carpentry is complaining that the images do not have alt-texts, so maybe also have a look at that?

**Comment by CunliangGeng** [comment]
@frankcorneliusmartin Thanks for the comments, I have pushed one new commit to apply your suggestions.

File: learners/setup.md (hunk 2 above)
Comment (bartvanb) [reply learners/setup.md]: @CunliangGeng I'm guessing you got the Ubuntu 18+ from our documentation? Then I guess we should also update there

Comment (bartvanb) [reply learners/setup.md]: I just checked and indeed docs were outdated. Included an update in PR https://github.com/vantage6/vantage6/pull/1329

Comment (CunliangGeng) [reply learners/setup.md]: Not only the docs but chapter 6 needs an [update](https://github.com/vantage6/vantage6-workshop/pull/29/commits/6ddd9dbcb51c8001cacc9a8a911cc4f47d286fef).

Comment (CunliangGeng) [reply learners/setup.md]: It looks a bit annoying to update the ubuntu version from time to time. Shall we change the requirements to e.g.  "Ubuntu ([version with standard support](https://wiki.ubuntu.com/Releases))"?

File: learners/setup.md [hunk 7]
```diff
@@ -2,53 +2,158 @@
...
-<!--
-FIXME: place any data you want learners to use in `episodes/data` and then use
-       a relative link ( [data zip file](data/lesson-data.zip) ) to provide a
-       link to it, replacing the example.com link.
--->
-Download the [data zip file](https://example.com/FIXME) and unzip it to your Desktop
+## Software setup
 
-## Software Setup
+This lesson requires the following software to be installed on your system:
```
Comment (bartvanb) [inline learners/setup.md:11]: I would say: "Vantage6 requires the following software to be installed on your system. In this lesson, we will explain how to install it"

File: learners/setup.md [hunk 8]
```diff
@@ -2,53 +2,158 @@
...
 
-::::::::::::::::::::::::::::::::::::::: discussion
+- Recommended operating system: Ubuntu 20.04+ , MacOS Big Sur+, or Windows 10+
+- Docker Desktop (always latest version)
+- Conda (latest version)
+- Python
+    - Python v3.10 for vantage6 version 3.8.0 or higher
```
Comment (bartvanb) [inline learners/setup.md:17]: I would shorten this to just say "Python v3.10" (no-one just starting to use vantage6 should use versions below 3.8 :-))

File: learners/setup.md [hunk 9]
```diff
@@ -2,53 +2,158 @@
...
-::::::::::::::::::::::::
+### Conda
 
+There are several ways to install Conda on your system. For this lesson, we recommend using Miniconda installer.
 
-:::::::::::::::: spoiler
+Follow the [official instructions](https://docs.anaconda.com/free/miniconda/#quick-command-line-install) to install Miniconda on Windows, MacOS, or Linux.
 
-### Linux
+Open a **new** terminal window and verify the installation by running the command:
```
Comment (bartvanb) [inline learners/setup.md:81]: Maybe clarify this even further by saying "After completing the miniconda installation, open a new terminal..."

**Review by bartvanb — CHANGES_REQUESTED** [review]
My main question is if you can add somewhere what participants should do in case of issues with the installation (I guess contact us? Maybe we can use a specific email address for workshop questions?)

Other than that I'm very happy with it - just some minor text suggestions!

**Comment by CunliangGeng** [comment]
> My main question is if you can add somewhere what participants should do in case of issues with the installation (I guess contact us? Maybe we can use a specific email address for workshop questions?)

We have contact person on the website (https://workshop2024.vantage6.ai/). But we could add more people to reduce Djura's burden.
